library/src/engine/engine.ts

```diff
@@ -836,7 +836,7 @@ export function load(...pluginsToLoad: DatastarPlugin[]) {
     return a.name.localeCompare(b.name)
   })
 
-  pluginRegexs = plugins.map((plugin) => RegExp(`^${plugin.name}([A-Z]|_|$)`))
+  pluginRegexs = plugins.map((plugin) => RegExp(`^${plugin.name}([A-Z-]|_|$)`))
```

Comment:
We want plugins to match `<pluginname>-` from the raw string, which normally becomes the plugin name followed by an upper case character after the dataset camelcasing, except in the instance that the first character of the key was not a letter, then it will stay a -.